### PR TITLE
Create ScenarioSourceController

### PR DESCRIPTION
**Pull request by bielik-b**

### CODE REVIEW

File: src/main/java/executor/api/controller/ScenarioSourceController.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+import executor.api.model.ScenarioDTO;
+import executor.api.service.ScenarioSourceQueueHandlerImpl;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.Optional;
+
+@RestController
+@RequestMapping("/scenario")
```
Comment (Serhx4) [inline src/main/java/executor/api/controller/ScenarioSourceController.java:11]: let's add /api/v1/ prefix for all api endpoints
```suggestion
@RequestMapping("/api/v1/scenario")
```

File: src/main/java/executor/api/controller/ScenarioSourceController.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+@RestController
+@RequestMapping("/api/v1/scenario")
+public class ScenarioSourceController {
+   private ScenarioSourceQueueHandlerImpl scenarioSourceQueueHandler;
+
+    public ScenarioSourceController(ScenarioSourceQueueHandlerImpl scenarioSourceQueueHandler) {
+        this.scenarioSourceQueueHandler = scenarioSourceQueueHandler;
+    }
+
+    @PostMapping("/add")
```
Comment (artemkedyk) [inline src/main/java/executor/api/controller/ScenarioSourceController.java:19]: you can specify an add/get/update/delete request only with an annotation without a path
```suggestion
    @PostMapping
```

File: src/main/java/executor/api/controller/ScenarioSourceController.java [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    public ResponseEntity<?> offerScenario(@RequestBody ScenarioDTO scenarioDTO){
+        scenarioSourceQueueHandler.offer(scenarioDTO);
+       return ResponseEntity.ok("Offered successfully");
+    }
+    @GetMapping("/get")
```
Comment (artemkedyk) [inline src/main/java/executor/api/controller/ScenarioSourceController.java:24]:
```suggestion
    @GetMapping
```

File: src/main/java/executor/api/controller/ScenarioSourceController.java [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    public ScenarioDTO getScenario(){
+        Optional<ScenarioDTO> optional = Optional.ofNullable(scenarioSourceQueueHandler.poll());
+        return optional.orElseThrow(IllegalArgumentException::new);
```
Comment (Serhx4) [inline src/main/java/executor/api/controller/ScenarioSourceController.java:27]: Hi, @bielik-b!
Let's try to use `ResponseStatusException` here
Here's short [guide ](https://www.baeldung.com/spring-response-status-exception)about it

**Review by Duskmage2009 — APPROVED** [review]

File: src/main/java/executor/api/controller/ScenarioSourceController.java [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+import executor.api.model.ScenarioDTO;
+import executor.api.service.ScenarioSourceQueueHandlerImpl;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.Optional;
+
+@RestController
+@RequestMapping("/api/v1/scenario")
```
Comment (Serhx4) [inline src/main/java/executor/api/controller/ScenarioSourceController.java:11]: What if we use here plural `scenarios` ?
```suggestion
@RequestMapping("/api/v1/scenarios")
```

File: src/main/java/executor/api/controller/ScenarioSourceController.java [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+   private ScenarioSourceQueueHandlerImpl scenarioSourceQueueHandler;
+
+    public ScenarioSourceController(ScenarioSourceQueueHandlerImpl scenarioSourceQueueHandler) {
+        this.scenarioSourceQueueHandler = scenarioSourceQueueHandler;
+    }
+
+    @PostMapping
+    public ResponseEntity<?> offerScenario(@RequestBody ScenarioDTO scenarioDTO){
+        scenarioSourceQueueHandler.offer(scenarioDTO);
+       return ResponseEntity.ok("Offered successfully");
```
Comment (Serhx4) [inline src/main/java/executor/api/controller/ScenarioSourceController.java:22]: Let's try to return 201 Created here
here's an [example](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/http/ResponseEntity.html)

**Review by Serhx4 — APPROVED** [review]

File: src/main/java/executor/api/controller/ScenarioSourceController.java [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+@RequestMapping("/api/v1/scenarios")
+public class ScenarioSourceController {
+   private final ScenarioSourceQueueHandlerImpl scenarioSourceQueueHandler;
+
+    public ScenarioSourceController(ScenarioSourceQueueHandlerImpl scenarioSourceQueueHandler) {
+        this.scenarioSourceQueueHandler = scenarioSourceQueueHandler;
+    }
+
+    @PostMapping
+    public ResponseEntity<?> offerScenario(@RequestBody ScenarioDTO scenarioDTO){
```
Comment (artemkedyk) [inline src/main/java/executor/api/controller/ScenarioSourceController.java:22]: let's specify type of generic

**Review by artemkedyk — APPROVED** [review]

**Pull request merged**